src/key.rs

```diff
@@ -608,6 +618,247 @@ impl HasByteSource for SecretAsymmetricKey {
     }
 }
 
+#[derive(Serialize, Deserialize, Debug)]
+pub enum SigningKey {
+    SodiumOxideEd25519(SodiumOxideEd25519SecretAsymmetricKey),
+    RingEd25519(RingEd25519SecretAsymmetricKey),
+}
+
+impl StorableType for SigningKey {}
+
+impl From<SigningKey> for Key {
+    fn from(signing_key: SigningKey) -> Self {
+        match signing_key {
+            SigningKey::SodiumOxideEd25519(k) =>
+                Key::Asymmetric(AsymmetricKey::Public(PublicAsymmetricKey::SodiumOxideEd25519(k.public_key().unwrap()))),
+            SigningKey::RingEd25519(k) =>
+                Key::Asymmetric(AsymmetricKey::Public(PublicAsymmetricKey::RingEd25519(k.public_key().unwrap())))
+        }
+    }
+}
+
+
+#[derive(Serialize, Deserialize, Debug)]
+pub enum EncryptingKey {
+    SodiumOxideCurve25519(SodiumOxideCurve25519SecretAsymmetricKey),
+    SodiumOxideSymmetricKey(SodiumOxideSymmetricKey),
+}
+
+impl StorableType for EncryptingKey {}
+
+#[derive(Serialize, Deserialize, Debug, Copy, Clone)]
+#[serde(tag = "t", content = "c")]
```

Comment:
Don't need serde tagging constraints here, this builder won't ever be serialized into an Entry, as it does not belong to the TypeBuilders enum.